components/tabs/src/storage.rs

```diff
@@ -399,6 +394,23 @@ fn compute_serialized_size(v: &Vec<RemoteTab>) -> usize {
     serde_json::to_string(v).unwrap_or_default().len()
 }
 
+// Similar to places/utils.js
+// handles the case where `s.is_char_boundary(max_len)`
+// is false (which would otherwise panic).
+pub fn slice_up_to(s: String, max_len: usize) -> String {
+    if max_len >= s.len() {
+        return s;
+    }
+    let mut idx = max_len;
+    while !s.is_char_boundary(idx) {
```

Comment:
Note: I'm pretty sure this check is O(1), and therefore this is a better method than iterating with `char_indices()` (
https://doc.rust-lang.org/src/core/num/mod.rs.html#882).

Reply:
Interestingly enough, `is_utf8_char_boundary()` gets called by `is_char_boundary` https://doc.rust-lang.org/src/core/str/mod.rs.html#211 so I think there's something there